airbyte_cdk/connector_builder/test_reader/helpers.py

```diff
@@ -718,3 +720,23 @@ def get_auxiliary_request_type(stream: dict, http: dict) -> str:  # type: ignore
     Determines the type of the auxiliary request based on the stream and HTTP properties.
     """
     return "PARENT_STREAM" if stream.get("is_substream", False) else str(http.get("type", None))
+
+
+def convert_state_blob_to_mapping(
```

Comment:
as mentioned in the description, during testing I noticed that the state value is always `{}`. After testing this a bit, it happens when we call `asdict()` in the handler/main which ends up zeroing the `AirbyteStateBlob` value. This solution is a bit meh and the original type in the above definition is slightly wrong because it isn't just a `Dict` it can be a state message as evidenced here